debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -224,14 +224,31 @@ private void processMessages(ChangeEventSourceContext context, PostgresPartition
                     offsetContext.updateWalPosition(lsn, lastCompletelyProcessedLsn, message.getCommitTime(), message.getTransactionId(), null,
                             taskContext.getSlotXmin(connection));
                     if (message.getOperation() == Operation.BEGIN) {
-                        dispatcher.dispatchTransactionStartedEvent(partition, Long.toString(message.getTransactionId()), offsetContext);
+                        dispatcher.dispatchTransactionStartedEvent(partition, message.getTransactionId().toString(), offsetContext);
```

Comment:
This is invoking `toString()` on `OptionalLong` now, which also causes that test failure of `shouldRegularlyFlushLsnWithTxMonitoring()`. I'll take care of it and also push some further smaller changes.